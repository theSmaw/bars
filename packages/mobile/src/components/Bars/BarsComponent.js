import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';

import { WithBarsContainer } from 'bars-shared';

import BarComponent from '../Bar/BarComponent';

export function BarsComponent({ bars, limit }) {
  return (
    <View>
      {
        bars.map((bar, index) => (
          <BarComponent

            // Generally considered an anti-pattern to use index as a key. We could massage the data
            // that comes back to add unique identifiers for each item, but this list is simple and
            // small and it is not worth the complexity
            key={index} // eslint-disable-line react/no-array-index-key

            limit={limit}
            value={bar}
          />
        ))
      }
    </View>
  );
}

BarsComponent.propTypes = {
  bars: PropTypes.arrayOf(PropTypes.number).isRequired,
  limit: PropTypes.number.isRequired
};

export default WithBarsContainer(BarsComponent);
