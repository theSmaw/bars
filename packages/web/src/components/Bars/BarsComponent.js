import React from 'react';
import PropTypes from 'prop-types';

import { WithBarsContainer } from 'bars-shared';

import BarComponent from '../Bar/BarComponent';

export function BarsComponent({
  bars,
  buttons,
  limit,
  onButtonClicked,
  onSelected,
  selectedBar
}) {
  return (
    <div className="bars">
      {
        bars.map((bar, index) => (
          <BarComponent

            // Generally considered an anti-pattern to use index as a key. We could massage the data
            // that comes back to add unique identifiers for each item, but this list is simple and
            // small and it is not worth the complexity
            key={index} // eslint-disable-line react/no-array-index-key

            limit={limit}
            value={bar}
          />
        ))
      }
      <div>
        <select
          className="barsSelect"
          onChange={
            /* istanbul ignore next */
            e => onSelected(e.target.value)
          }
          value={selectedBar}
        >
          {
            bars.map((bar, index) => (
              <option
                key={index} // eslint-disable-line react/no-array-index-key
                value={index}
              >
                Bar {index + 1}
              </option>
            ))
          }
        </select>
        <div>
          {
            buttons.map((button, index) => (
              <button
                className="barsButton"
                key={index} // eslint-disable-line react/no-array-index-key
                onClick={
                  /* istanbul ignore next */
                  () => onButtonClicked(button)
                }
              >
                {button}
              </button>
            ))
          }
        </div>
      </div>
    </div>
  );
}

BarsComponent.propTypes = {
  bars: PropTypes.arrayOf(PropTypes.number).isRequired,
  buttons: PropTypes.arrayOf(PropTypes.number).isRequired,
  limit: PropTypes.number.isRequired,
  onButtonClicked: PropTypes.func.isRequired,
  onSelected: PropTypes.func.isRequired,
  selectedBar: PropTypes.number.isRequired
};

export default WithBarsContainer(BarsComponent);
