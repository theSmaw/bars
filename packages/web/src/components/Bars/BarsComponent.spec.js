import React from 'react';
import { shallow } from 'enzyme';

import { BarsComponent } from './BarsComponent';

describe('BarsComponent', () =>

  it('Should render bars', () => {
    const barsComponent = shallow(
      <BarsComponent
        bars={[1, 2, 3]}
        buttons={[10, 20, -30]}
        limit={50}
        onButtonClicked={() => {}}
        onSelected={() => {}}
        selectedBar={0}
      />
    );

    expect(barsComponent).toMatchSnapshot();
  })
);
