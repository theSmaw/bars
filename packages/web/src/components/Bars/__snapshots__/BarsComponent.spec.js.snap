// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarsComponent Should render bars 1`] = `
<div
  className="bars"
>
  <BarComponent
    key="0"
    limit={50}
    value={1}
  />
  <BarComponent
    key="1"
    limit={50}
    value={2}
  />
  <BarComponent
    key="2"
    limit={50}
    value={3}
  />
  <div>
    <select
      className="barsSelect"
      onChange={[Function]}
      value={0}
    >
      <option
        key="0"
        value={0}
      >
        Bar 
        1
      </option>
      <option
        key="1"
        value={1}
      >
        Bar 
        2
      </option>
      <option
        key="2"
        value={2}
      >
        Bar 
        3
      </option>
    </select>
    <div>
      <button
        className="barsButton"
        key="0"
        onClick={[Function]}
      >
        10
      </button>
      <button
        className="barsButton"
        key="1"
        onClick={[Function]}
      >
        20
      </button>
      <button
        className="barsButton"
        key="2"
        onClick={[Function]}
      >
        -30
      </button>
    </div>
  </div>
</div>
`;
